packages/firestore/src/remote/watch_change.ts

```diff
@@ -546,33 +611,33 @@ export class WatchChangeAggregator {
   }
 
   /**
-   * Resets the initial state of a Watch target to its initial state (e.g. sets
+   * Resets the state of a Watch target to its initial state (e.g. sets
    * 'current' to false, clears the resume token and removes its target mapping
    * from all documents).
    */
   private resetTarget(targetId: TargetId): void {
+    assert(
+      !this.targetStates[targetId].isPending,
+      'Should only reset active targets'
+    );
     delete this.targetStates[targetId];
```

Comment:
I feel like we shouldn't completely delete the targetStates entry (or else we should immediately recreate it).  It may not be an issue because the removeDocumentFromTarget() calls below end up re-creating it, but I'm wondering if there's any case where that doesn't happen and as a result we don't include the target in our next RemoteEvent.  E.g. this could happen if we got an existence filter mismatch for an empty target (which would result in us failing to raise a fromCache=true event), except I don't think that's possible...

Since I can't actually come up with a broken case, I leave it up to you to decide if there's an issue warranting any change here.